### PR TITLE
fix(packages): set requestStates in case of wrong auth

### PR DESCRIPTION
**Pull request by MarkusAzer**
#### Summary
fix set requestStates in case of wrong auth

#### Description

Package : sdk-middleware-auth
fix set requestStates in case of wrong auth

This PR is update for [1279](https://github.com/commercetools/nodejs/issues/1279)

### CODE REVIEW

**Comment by MarkusAzer** [comment]
Thanks @daern91  :)

You right and i will add test.

File: packages/sdk-middleware-auth/src/base-auth-flow.js [hunk 1]
```diff
@@ -99,10 +99,17 @@ function executeRequest({
...
         if (parsed) error.body = parsed
+
+        // Dispatch all pending requests
+        requestState.set(false)
+
         response.reject(error)
       })
     })
     .catch((error: Error) => {
+      // Dispatch all pending requests
```
Comment (daern91) [inline packages/sdk-middleware-auth/src/base-auth-flow.js]: same question here

File: packages/sdk-middleware-auth/src/base-auth-flow.js [hunk 2]
```diff
@@ -99,10 +99,17 @@ function executeRequest({
         }
         const error: Object = new Error(parsed ? parsed.message : text)
         if (parsed) error.body = parsed
+
+        // Dispatch all pending requests
```
Comment (daern91) [inline packages/sdk-middleware-auth/src/base-auth-flow.js]: I'm not understanding this comment, it looks like we are just rejecting below and not actually dispatching any requests?

Comment (MarkusAzer) [reply packages/sdk-middleware-auth/src/base-auth-flow.js]: There is queue of requests,
For ex, once 1st request is failed, then requestState set to false, and dispatch all other queued requests
so 2nd request starts and so on

Comment (daern91) [reply packages/sdk-middleware-auth/src/base-auth-flow.js]: I can see that we do that in the condition above: https://github.com/commercetools/nodejs/blob/192f8d02a657bc500ffb0b8845f80bcb1aeadad2/packages/sdk-middleware-auth/src/base-auth-flow.js#L82-L88

but are you sure we also keep dispatching here? I cant see it in the code so maybe I'm just missing a part of the logic somewhere.

Comment (MarkusAzer) [reply packages/sdk-middleware-auth/src/base-auth-flow.js]: You right, it should be then sth like reset request status!

File: packages/sdk-middleware-auth/src/base-auth-flow.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ function executeRequest({
                 tokenCacheKey
               )
 
-              // Dispatch all pending requests
+              // Reset requestState to false
```
Comment (daern91) [inline packages/sdk-middleware-auth/src/base-auth-flow.js:74]: we shouldn't change this comment

File: packages/sdk-middleware-auth/src/base-auth-flow.js [hunk 4]
```diff
@@ -99,10 +99,17 @@ function executeRequest({
         }
         const error: Object = new Error(parsed ? parsed.message : text)
         if (parsed) error.body = parsed
+
+        // Reset requestState to false
```
Comment (daern91) [inline packages/sdk-middleware-auth/src/base-auth-flow.js:103]: Could we please make this comment a bit more descriptive? I.e., why are we resetting the request state? 

That we are setting it to false is already described well by the code already so currently it's a superfluous comment IMO.

File: packages/sdk-middleware-auth/test/base-auth-flow.spec.js [hunk 5]
```diff
@@ -371,6 +371,53 @@ describe('Base Auth Flow', () => {
...
+    const requestState = store(false)
+    const pendingTasks = []
+
+    const startCreateBaseMiddleware = () =>
+      new Promise((resolve, reject) => {
+        expect(requestState.get()).toBe(false)
+
+        // fire off promise returning function and change requestState to true
+        // we do not await the response before the assertion since we set
+        // reqeustState to false in the catch block
```
Comment (daern91) [inline packages/sdk-middleware-auth/test/base-auth-flow.spec.js:400]: nitpick: I checked the code now and technically we don't set it in the catch block but here: https://github.com/commercetools/nodejs/blob/acf7cd4a4b208736dee67a6b946a8e7ccbd07eb8/packages/sdk-middleware-auth/src/base-auth-flow.js#L92-L107

File: packages/sdk-middleware-auth/test/base-auth-flow.spec.js [hunk 6]
```diff
@@ -371,6 +371,53 @@ describe('Base Auth Flow', () => {
...
+        })
+
+        expect(requestState.get()).toBe(true)
+        expect(requestCount).toBe(1) // Make sure that nock runned the mocked request
+
+        return start
+      })
+
+    try {
+      // await promise to fail and set requestState to false
```
Comment (daern91) [inline packages/sdk-middleware-auth/test/base-auth-flow.spec.js:414]:
```suggestion
      // await promise failure which should set `requestState` to `false`
```

**Pull request merged**